### PR TITLE
add eachWeekOfInterval

### PR DESCRIPTION
**Pull request by xkizer**
See #540

### CODE REVIEW

File: src/eachWeekOfInterval/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * @name eachWeekOfInterval
+ * @category Interval Helpers
+ * @summary Return the array of weeks within the specified time interval.
+ *
+ * @description
+ * Return the array of weeks within the specified time interval.
+ *
+ * @param {Interval} interval - the interval. See [Interval]{@link docs/types/Interval}
+ * @param {Options} [options] - the object with options. See [Options]{@link https://date-fns.org/docs/Options}
+ * @param {0|1|2} [options.additionalDigits=2] - passed to `toDate`. See [toDate]{@link https://date-fns.org/docs/toDate}
```
Comment (leshakoss) [inline src/eachWeekOfInterval/index.js]: Please add ` * @param {0|1|2|3|4|5|6} [options.weekStartsOn=0] - the index of the first day of the week (0 - Sunday)`

File: src/eachWeekOfInterval/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  var endDate = toDate(interval.end, dirtyOptions)
+
+  var endTime = endDate.getTime()
+
+  // Throw an exception if start date is after end date or if any date is `Invalid Date`
+  if (!(startDate.getTime() <= endTime)) {
+    throw new RangeError('Invalid interval')
+  }
+
+  var startDateWeek = startOfWeek(startDate)
```
Comment (leshakoss) [inline src/eachWeekOfInterval/index.js]: Please pass `options` to this function, as in `startOfWeek(startDate, dirtyOptions)`

File: src/eachWeekOfInterval/test.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      new Date(2014, 9 /* Oct */, 12),
+      new Date(2014, 9 /* Oct */, 19),
+      new Date(2014, 9 /* Oct */, 26),
+      new Date(2014, 10 /* Nov */, 2),
+      new Date(2014, 10 /* Nov */, 9),
+      new Date(2014, 10 /* Nov */, 16),
+      new Date(2014, 10 /* Nov */, 23)
+    ])
+  })
+
```
Comment (leshakoss) [inline src/eachWeekOfInterval/test.js]: Please add a `allows to specify which day is the first day of the week` test, like this: https://github.com/date-fns/date-fns/blob/master/src/endOfWeek/test.js#L16-L22

The functionality should be already implemented by passing `options` to `startOfWeek`

File: src/eachWeekOfInterval/test.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  it('throws an exception if the interval is undefined', function () {
+    var block = eachWeekOfInterval.bind(
+      null,
+      // $ExpectedMistake
+      undefined
+    )
+    assert.throws(block, RangeError)
+  })
+
```
Comment (leshakoss) [inline src/eachWeekOfInterval/test.js]: Please add ```throws `RangeError` if `options.weekStartsOn` is not convertable to 0, 1, ..., 6 or undefined``` test, like this: https://github.com/date-fns/date-fns/blob/master/src/endOfWeek/test.js#L125-L129

File: src/fp/eachWeekOfIntervalWithOptions/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// This file is generated automatically by `scripts/build/typings.js`. Please, don't change it.
+
+import {eachWeekOfIntervalWithOptions} from 'date-fns/fp'
+export = eachWeekOfIntervalWithOptions
```
Comment (leshakoss) [inline src/fp/eachWeekOfIntervalWithOptions/index.d.ts]: Please add `eachWeekOfInterval` and `eachWeekOfIntervalWithOptions` tests to src/fp/test.js

File: CHANGELOG.md [hunk 6]
```diff
@@ -15,6 +15,8 @@ for the list of changes made since `v2.0.0-alpha.1`.
 
 ### Added
 
+- Added `eachWeekOfInterval`, the weekly equivalent of `eachDayOfInterval`
+
```
Comment (leshakoss) [inline CHANGELOG.md]: Please move the changelog entry to the bottom of `Added` list :-)

**Comment by kossnocorp** [comment]
This PR is released as `2.0.0-alpha.11` 🙏

**Pull request merged**